### PR TITLE
Refs #30630 - upgrade 2to3 migration gem

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

File: katello.gemspec [hunk 1]
```diff
@@ -55,7 +55,7 @@ Gem::Specification.new do |gem|
   gem.add_dependency "pulp_ansible_client", "> 0.2", "< 0.3"
   gem.add_dependency "pulp_container_client", ">= 2.0.0", "< 2.1.0"
   gem.add_dependency "pulp_rpm_client", ">=3.6.0", "< 3.7.0"
-  gem.add_dependency "pulp_2to3_migration_client", ">= 0.2.0b6", "< 0.3.0"
+  gem.add_dependency "pulp_2to3_migration_client", ">= 0.3.0", "< 0.4.0"
```
Comment (parthaa) [inline katello.gemspec:58]:
```suggestion
  gem.add_dependency "pulp_2to3_migration_client", ">= 0.3.0", "<= 0.4.0"
```

**Comment by jlsherrill** [comment]
[test katello]

**Comment by jlsherrill** [comment]
pushed a vcr update which got missed in the main 3.6 update because a pr added it after the tests were run

**Comment by jlsherrill** [comment]
[test katello]

**Pull request merged**